containers/daap-python-base/tests/unit/data_product_metadata_test.py

```diff
@@ -31,3 +81,58 @@ def test_get_specific_metadata_spec_path(monkeypatch):
         path
         == "s3://bucket/data_product_metadata_spec/v1/moj_data_product_metadata_spec.json"
     )
+
+
+def test_metadata_exist(s3_client, region_name, monkeypatch):
+    bucket_name = "bucket"
+    setup_bucket(bucket_name, s3_client, region_name, monkeypatch)
+
+    # populate some folders & files to mock s3 bucket
+    file_text = json.dumps(test_schema_pass)
+    with NamedTemporaryFile(delete=True, suffix=".json") as tmp:
+        with open(tmp.name, "w", encoding="UTF-8") as f:
+            f.write(file_text)
+
+        s3_client.upload_file(
+            tmp.name, bucket_name, "metadata/test_product/v1.0/metadata.json"
+        )
+
+    md = DataProductMetadata(test_schema_pass["name"], logging.getLogger())
+    assert md.metadata_exists
```

Comment:
You might have to monkeypatch the `s3_client` in the `data_product_metadata`, since it is a different instance to the one you are uploading the file to above.

A pattern I've used elsewhere is to make DataProductMetadata take an optional s3_client parameter that defaults to the module global value. That way you can pass it in from the tests.